lms/djangoapps/mobile_api/social_facebook/friends/tests.py

```diff
@@ -0,0 +1,394 @@
+        self.assertTrue('id' in response.data['friends'][0] and response.data['friends'][0]['id'] == self.FB_ID_1)
+        self.assertTrue('name' in response.data['friends'][0] and response.data['friends'][0]['name'] == self.USERNAME_1)
+        # Assert that USERNAME_2 is returned
+        self.assertTrue('id' in response.data['friends'][1] and response.data['friends'][1]['id'] == self.FB_ID_2)
+        self.assertTrue('name' in response.data['friends'][1] and response.data['friends'][1]['name'] == self.USERNAME_2)
+        # Assert that USERNAME_3 is returned
+        self.assertTrue('id' in response.data['friends'][2] and response.data['friends'][2]['id'] == self.FB_ID_3)
+        self.assertTrue('name' in response.data['friends'][2] and response.data['friends'][2]['name'] == self.USERNAME_3)
+
+
+    '''
+        Helper Functions 
+    '''
+
+    def set_facebook_interceptor(self, data): 
+        httpretty.register_uri(httpretty.GET, 
+                        "https://graph.facebook.com/v2.2/me/friends",
+                        body=json.dumps(data),
+                        status=201)
+
+    def user_create_and_signin(self, user_number): 
+        if user_number == 1:
+            self.user_1 = UserFactory.create(   username=self.USERNAME_1, 
+                                                email=self.EMAIL_1, 
+                                                password=self.PASSWORD_1)
+            self.client.login(username=self.USERNAME_1, password=self.PASSWORD_1)
+        elif user_number == 2:
+            self.user_2 = UserFactory.create(   username=self.USERNAME_2, 
+                                                email=self.EMAIL_2, 
+                                                password=self.PASSWORD_2)
+            self.client.login(username=self.USERNAME_2, password=self.PASSWORD_2)
+        elif user_number == 3:
+            self.user_3 = UserFactory.create(   username=self.USERNAME_3, 
+                                                email=self.EMAIL_3, 
+                                                password=self.PASSWORD_3)
+            self.client.login(username=self.USERNAME_3, password=self.PASSWORD_3)
+
+    def enroll_in_course(self, user, course): 
+        resp = self._change_enrollment('enroll')
+        self.assertEqual(resp.status_code, 200)
+        self.assertTrue(CourseEnrollment.is_enrolled(user, course.id))
+        course_mode, is_active = CourseEnrollment.enrollment_mode_for_user(user, course.id)
+        self.assertTrue(is_active)
+        self.assertEqual(course_mode, 'honor')
+
+    def link_edX_account_to_social_backend(self, user, backend, social_uid):
+        self.url = reverse(login_oauth_token, kwargs={"backend": backend})
+        # self.social_uid = "10155110991745300"
+        UserSocialAuth.objects.create(user=user, provider=backend, uid=social_uid)
+
+    def format_course_id(self): 
+        return self.course.scope_ids.usage_id.course_key._to_string().replace('+', '/')
```

Comment:
Yes the course id needs to be passed in the format: course_school/course_name/course_run
If there is an easier way of  deriving this format from the SlashSeparatedCourseKey it would be great if you could point me in the right direction.